engine/installation/linux/ubuntulinux.md

```diff
@@ -116,365 +117,430 @@ packages from the Docker repository:
+| `xserver-xorg-lts-trusty`         | Optional in non-graphical environments without Unity/Xorg. **Required** when running Docker on machine with a graphical environment. |
+| `ligbl1-mesa-glx-lts-trusty`      | To learn more about the reasons for these packages, read the installation instructions for backported kernels, specifically the [LTS Enablement Stack](https://wiki.ubuntu.com/Kernel/LTSEnablementStack). Refer to note 5 under each version. |
+
 
 To upgrade your kernel and install the additional packages, do the following:
 
-1. Open a terminal on your Ubuntu host.
+1.  Open a terminal on your Ubuntu host.
 
-2. Update your package manager.
+2.  Update your package manager.
 
-        $ sudo apt-get update
+    ```bash
+    $ sudo apt-get update
+    ```
 
-3. Install both the required and optional packages.
+3.  Install both the required and optional packages.
 
-        $ sudo apt-get install linux-image-generic-lts-trusty
+    ```bash
+    $ sudo apt-get install linux-image-generic-lts-trusty
+    ```
 
-    Depending on your environment, you may install more as described in the preceding table.
+    Repeat this step for other packages you need to install.
 
-4. Reboot your host.
+4.  Reboot your host to use the updated kernel.
 
-        $ sudo reboot
+    ```bash
+    $ sudo reboot
+    ```
 
-5. After your system reboots, go ahead and install Docker.
+5.  After your system reboots, go ahead and
+    [install Docker](ubuntulinux.md#install).
 
 ## Install
 
-Make sure you have installed the prerequisites for your Ubuntu version.
-
-Then,
-install Docker using the following:
+Make sure you have satisfied all the
+[prerequisites](ubuntulinux.md#prerequisites), then follow these steps.
 
-1. Log into your Ubuntu installation as a user with `sudo` privileges.
+1.  Log into your Ubuntu installation as a user with `sudo` privileges.
 
-2. Update your `APT` package index.
+2.  Update your `APT` package index.
 
-        $ sudo apt-get update
-
-3. Install Docker.
+    ```bash
+    $ sudo apt-get update
+    ```
+3.  Install Docker.
 
-        $ sudo apt-get install docker-engine
+    ```bash
+    $ sudo apt-get install docker-engine
+    ```
 
-4. Start the `docker` daemon.
+4.  Start the `docker` daemon.
 
-        $ sudo service docker start
+    ```bash
+    $ sudo service docker start
+    ```
 
-5. Verify `docker` is installed correctly.
+5.  Verify that `docker` is installed correctly by running the `hello-world`
+    image.
 
-        $ sudo docker run hello-world
+    ```bash
+    $ sudo docker run hello-world
+    ```
 
     This command downloads a test image and runs it in a container. When the
-    container runs, it prints an informational message. Then, it exits.
+    container runs, it prints an informational message and exits.
 
```

Comment:
One question that keeps popping up is "how to install a specific version". Do you think we should add a short section after this explaining that?

So far we considered it "general knowledge" when using `apt`, but perhaps it's good to show.

For _production_, it's also recommended to pin to a specific version, to prevent inadvertently upgrading docker to a new version (which should never go without thought).